### PR TITLE
add discord verify, change prisma schema

### PR DESCRIPTION
**Pull request by Karik-ribasu**

### CODE REVIEW

File: src/pages/api/quest/oauth/discord.ts [hunk 1]
```diff
@@ -70,7 +97,7 @@ async function prismaInsertUserDiscordData(
     const userDiscordData = await prismaDB.userDiscordData.create({
       data: {
         userId: userId,
-        discordId: discordId,
+        discordId: 'teste id 123',
```
Comment (rodrigopavezi) [inline src/pages/api/quest/oauth/discord.ts:100]: @Karik-ribasu we need to revert this.

File: src/pages/api/quest/oauth/verify-discord.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+import { Prisma, PrismaClient } from '@prisma/client';
+import querystring from 'querystring';
+
+type Response = {};
+
+async function hasUserGuildAndRole(
+  code: string,
+  userDiscordId: string | undefined
+) {
+  const clientId = process.env.DISCORD_CLIENT_ID;
```
Comment (rodrigopavezi) [inline src/pages/api/quest/oauth/verify-discord.ts:11]: Here we can reuse this key NEXT_PUBLIC_DISCORD_CLIENT_ID

File: src/pages/api/quest/oauth/verify-discord.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  const clientSecret = process.env.DISCORD_CLIENT_SECRET;
+  const redirectUri = process.env.DISCORD_REDIRECT_URI;
```
Comment (rodrigopavezi) [inline src/pages/api/quest/oauth/verify-discord.ts:13]: we need to get the redirect url from the frontend as its from env to env

File: src/pages/api/quest/oauth/verify-discord.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+  };
+
+  const requestBody = querystring.stringify(requestData);
+
+  const memberResponse = await fetch(
+    `https://discord.com/api/v10/guilds/${guildId}/members/${userDiscordId}`,
+    {
+      method: 'GET',
+      headers: {
+        Authorization: `Bot ${process.env.DISCORD_BOT_TOKEN}`,
```
Comment (rodrigopavezi) [inline src/pages/api/quest/oauth/verify-discord.ts:31]: if we have the code from the user why we need a BOT token here?

**Review by rodrigopavezi — CHANGES_REQUESTED** [review]

File: src/pages/api/quest/oauth/discord.ts (hunk 1 above)
Comment (Karik-ribasu) [reply src/pages/api/quest/oauth/discord.ts]: fixed.

File: src/pages/api/quest/oauth/verify-discord.ts (hunk 2 above)
Comment (Karik-ribasu) [reply src/pages/api/quest/oauth/verify-discord.ts:12]: fixed.

File: src/pages/api/quest/oauth/verify-discord.ts (hunk 3 above)
Comment (Karik-ribasu) [reply src/pages/api/quest/oauth/verify-discord.ts]: fixed.

File: src/pages/api/quest/oauth/verify-discord.ts (hunk 4 above)
Comment (Karik-ribasu) [reply src/pages/api/quest/oauth/verify-discord.ts:31]: we need this token to get data from user as a member and read their roles in the guild.

**Review by rodrigopavezi — APPROVED** [review]
looks good 👍

**Pull request merged**